node/node.go

```diff
@@ -31,6 +32,21 @@ const (
 // CalculateNextEpoch is a function used to calculate the next `SendEpoch` for a given message.
 type CalculateNextEpoch func(count uint64, epoch int64) int64
 
+type EventStatus int
+
+const (
+	ONLINE EventStatus = iota
```

Comment:
minor:
const are `MixedCase` in go
``
Constant names must use [MixedCaps](https://google.github.io/styleguide/go/guide#mixed-caps) like all other names in Go. ([Exported](https://tour.golang.org/basics/3) constants start with uppercase, while unexported constants start with lowercase.)
``

https://google.github.io/styleguide/go/decisions.html

Maybe you want to also use something a bit more specific than online/offline `OnlineStatus`/`OfflineStatus`